index.js

```diff
@@ -1,50 +1,62 @@
 'use strict'
 
-const binary = require('@mapbox/node-pre-gyp')
-const path = require('path')
-const pkgpath = path.resolve(path.join(__dirname, './package.json'))
-const binding_path = binary.find(pkgpath) // eslint-disable-line camelcase
-const bindings = require(binding_path)
-
-module.exports = bindings
-
-module.exports.version = require('./package.json').version
-module.exports.init = function (sk) {
-  return module.exports.oboeInit({ serviceKey: sk || process.env.SW_APM_SERVICE_KEY })
+function platform () {
+  const cpu = process.arch
+  const os = process.platform
+  const node = process.version.split('.')[0].slice(1)
+
+  if (os === 'linux') {
+    const glibc = process.report.getReport().header.glibcVersionRuntime
+    const abi = glibc ? 'gnu' : 'musl'
+
+    return `${cpu}-linux-${abi}-${node}`
+  } else {
+    return `${cpu}-${os}-${node}`
+  }
 }
+const p = platform()
+
+let bindings
+let metrics
 
 try {
-  // this is hardcoded as node-pre-gyp doesn't know about multiple targets.
-  const metrics_path = binding_path.replace('apm_bindings.node', 'ao_metrics.node') // eslint-disable-line camelcase
-  module.exports.metrics = require(metrics_path)
-} catch (e) {
-  // eslint-disable-next-line no-console
-  console.warn(`metrics disabled ${e.message}`)
-  // return an dummy metrics if this can't be loaded
-  module.exports.metrics = {
-    start () { return true },
-    stop () { return true },
-    getMetrics () { return {} }
+  bindings = require(`./npm/${p}/bindings.node`)
+  metrics = require(`./npm/${p}/metrics.node`)
+} catch {
+  try {
+    bindings = require(`solarwinds-apm-bindings-${p}/bindings.node`)
+    metrics = require(`solarwinds-apm-bindings-${p}/metrics.node`)
+  } catch {
+    console.warn(`platform ${p} not supported`)
```

Comment:
from my understanding and some testing this was to support running on non-LTS node versions where the main bindings would work (since they use Node-API which is version independent) but the metrics wouldn't (since they use nan which isn't). given we don't support non-LTS to begin with and doing so would explode the number of packages to publish (and no one should be using non-LTS in prod to begin with) it felt reasonable to kill that functionality

Reply:
the other option would be to publish the bindings and metrics as separate packages with the bindings being a single version-agnostic package per os/cpu/libc that could be used on non-LTS and the metrics being os/cpu/libc/version dependent without LTS support